quarkchain/env.py

```diff
@@ -1,14 +1,16 @@
 import copy
 from quarkchain.config import get_default_evm_config, QuarkChainConfig
+from quarkchain.constants import PRECOMPILED_CONTRACTS_AFTER_EVM_ENABLED
 from quarkchain.evm.config import Env as EvmEnv
 from quarkchain.db import InMemoryDb
 from quarkchain.cluster.cluster_config import ClusterConfig
+from quarkchain.evm.specials import specials
 
 
 class Env:
-    def __init__(self, db=None, evm_config=None, cluster_config=None):
+    def __init__(self, db=None, evm_config=None):
         self.db = db or InMemoryDb()
-        self.cluster_config = cluster_config if cluster_config else ClusterConfig()
+        self.__cluster_config = ClusterConfig()
```

Comment:
Why not still use cluster_config and then apply timestamp in the ctor?  This makes sure we have the best backward-compatible interface.

Reply:
some places we are manually setting the cluster config (like in `slave.py`):

```python
    env = DEFAULT_ENV.copy()
    env.cluster_config = ClusterConfig.create_from_args(args)
    env.slave_config = env.cluster_config.get_slave_config(args.node_id)
    env.arguments = args
```

Reply:
Then I would suggest using

self.cluster_config = ClusterConfig()